src/test/java/htsjdk/samtools/util/AsyncWriterTest.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2016 Len Trigg
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package htsjdk.samtools.util;
+
+import org.testng.Assert;
+import org.testng.annotations.Test;
+
+public class AsyncWriterTest {
+    private static class MyException extends RuntimeException {
+        public MyException(String message) {
+            super(message);
+        }
+    }
+    private static class TestAsyncWriter extends AbstractAsyncWriter<Integer> {
+        protected TestAsyncWriter() {
+            super(500);
+        }
+
+        @Override
+        protected String getThreadNamePrefix() {
+            return "TestAsyncWriter";
+        }
+
+        @Override
+        protected void synchronouslyWrite(Integer item) {
+            if (item > 10) {
+                throw new MyException("InnerException");
+            }
+        }
+
+        @Override
+        protected void synchronouslyClose() {
+            // Nothing
+        }
+    }
+    @Test
+    public void testNoSelfSuppression() {
+        try {
+            try (TestAsyncWriter t = new TestAsyncWriter()) {
+                for (int i = 0; i < 1000; i++) {
+                    t.write(i);
+                }
+                Assert.fail("Expected exception");
+            }
+        } catch (Exception e) {
+            Assert.assertFalse(e.getMessage().contains("Self-suppression"));
```

Comment:
This seems redundant, and bad practice to assert based on the text of exception message. The `e instanceof MyException` assert below should be sufficient.